challenge_6/scala/ndsmith3/Ranges.scala

```diff
@@ -0,0 +1,34 @@
+import scala.annotation.tailrec
+
+object Ranges {
+  @tailrec
+  def makeRangeString(answer: List[String])(curr: Array[Int]): List[String] = {
+    val answerPlusNext       = (curr.head.toString :: answer)
+    lazy val nextIsIncrement = curr.head + 1 != curr(1)
+    lazy val lastIsDecrement = answer.head.toInt + 1 == curr.head
+    curr.length match {
+      case 1                      => if (lastIsDecrement) answerPlusNext.reverse else answer.reverse
```

Comment:
Instead of the nested if I would use two guarded cases like you used for the other 2 cases below to be consistent.